### DISCUSSION TITLE
not immediatly intuitive.

### DISCUSSION THREAD

**Discussion by emileswain** [postmanlabs/postman-flows#284] [Defect]
I'll admit i've not spent much time looking into the documentation, but to be honest i really shouldn't have to with a node based editor. 

So i'm just sharing some initial impressions/thoughts on where I got stuck and have had to leave the node editor to read up on how to do something, rather than it being imediately obviously.

1. There should be a node to set the environment the flow will use for all requests.
2. There should be a node to override environment variables scoped to the flow. 
3. Its not clear how nodes will use environment variables. For example, a node that uses a variable scoped to the collection, doesn't use the initial value, its just blank, as though the current value has to be set. What should the behaviour be?
4. There should be a node to easily define and set flow-variables that i want to pass into nodes to override variables scoped just to that node.
6. The node shows a triangle with an exclamation mark and talks about an Example. Yet the node runs ok. The exclamation mark suggests there is somthing wrong with the node. If its a suggestion, it should be a ? or something else. Not an error. 
7. Generally, i find the node parameters and layout just doesn't really make sense, so i'm having to go off and find documentation. I'm familiar with node programms like Nodred, VVVV, VDMX, which a much more complicated yet easy to use. 
8. The (mouse wheel) zoom delta has two levels. Too far and Too close.
9. Setting the variables in a node. I'm have no idea if these are environment, collection, node scoped. I want to set the {{my_variable}} value used by the request. It would help if the variable drop down, showed the variables the request is using. The intelisense for a variables value, doesn't make sense, it looks like its trying to set the value based on the request response. In which case, how do i set the value of a variable, before the request is made?
10. 

Generally, i find the node parameters and layout just doesn't really make sense, so i'm having to go off and find documentation. I'm familiar with node programms like Nodred, VVVV, VDMX, which a much more complicated yet easy to use.

**Comment by emileswain** [comment]
Came back to flows and i still don't find it as intuitive as i think it should be. 

An article showing how to work with environmental variables that are set as a result of a test that triggers the next block would be great. 

Its also not clear how taking the request output and piping it into the variables input should be treated or processed. 
My initial impression is that the variables input should essentially be either be the parameters that the request defines in its query/body parameters. or the variables should be the environmental variables, subsequently used in the request. 

WHich is it? or both?


If the request is the full request with header and body informaiton, i can grasp that. But what i can't grasp is how to transform that data into something to pass into the input variables of the next block. I've seen the demo where you define in the request something like {{body.mydata.avalue}} but i'm using environmental variables {{my_environment_variable}} and i do not want to couple a request into a specific incoming body response structure. 

As such its not clear in flows how best to go about pulling data from a response and packaging it up to pass to the next request. I'm guessing that i have use Create Durables or Create Data blocks, but they make zero sense and the suggested prompts make it even more confusing as to how to use them.

I've given up again. I'd rather script flows in objective C++

**Comment by emileswain** [comment]
https://www.postman.com/postman/workspace/example-flows/flow/6267f93cd0fb400392442ac8
This helped as well as the fact that you need to create examples for each request for the flow blocks to properly work.

**Comment by samay-v** [comment]
Thanks for all the feedback @emileswain! 
Last few months we've been working on a few improvements around issues you've raised:
1. Making it more intuitive to pass data between 2 requests
2. making it more apparent which scope is currently fulfilling the variable 
3. Making more content available for you if you get stuck somewhere.

I think this will solve a few of the issues you've been facing for long. I'll let you know here when these improvements are out on production so that you can test them out
